src/pages/workspaces/workspace/jobHistory/CallCacheWizard.js

```diff
@@ -0,0 +1,270 @@
+      else setDiffError(error)
+    }
+  }
+
+  const otherCallFqnSelectionOptions = _.flow(
+    _.keys,
+    _.map(name => { return { value: name, label: name } })
+  )
+
+  const resetDiffResult = () => {
+    setDiff(undefined)
+    setDiffError(undefined)
+  }
+
+  const resetCallSelection = () => {
+    resetDiffResult()
+    setOtherCallFqn(undefined)
+    setOtherIndex(undefined)
+    setOtherAttemptSucceeded(undefined)
+    setOtherCallSelected(false)
+  }
+
+  const resetWorkflowSelection = () => {
+    resetCallSelection()
+    setOtherWorkflowId(undefined)
+    setMetadataFetchError(undefined)
+  }
+
+  /*
+   * Page render
+   */
+
+  const divider = hr({ style: { width: '100%', marginTop: '1rem', marginBottom: '1rem', border: '1px ridge lightgray' } })
+
+  const step1 = () => {
+    return h(Fragment, [
+      div({ style: { paddingTop: '0.5rem', fontSize: 16, fontWeight: 500, ...Style.noWrapEllipsis } }, ['Step 1: Select the workflow you expected to cache from']),
+      div({ style: { marginTop: '0.5rem', marginBottom: '1rem', display: 'flex', flexDirection: 'row', alignItems: 'center' } }, [
+        div({ style: { paddingRight: '0.5rem' } }, ['Workflow ID:']),
+        div({ style: { paddingRight: '0.5rem', flex: '2 1 auto' } }, [h(TextInput, { style: Style.codeFont, id: 'otherWorkflowId' })]),
+        div([h(ButtonPrimary, {
+          style: { float: 'right' },
+          onClick: () => {
+            const otherWf = document.getElementById('otherWorkflowId').value
+            setOtherWorkflowId(otherWf)
+            readCalls(otherWf)
+          }
+        }, ['Continue >'])])
+      ]),
+      div({ style: { paddingTop: '0.5rem', fontSize: 16, fontWeight: 500, color: 'lightgray', ...Style.noWrapEllipsis } }, ['Step 2: Select the call you expected to cache from']),
+      divider,
+      div({ style: { paddingTop: '0.5rem', fontSize: 16, fontWeight: 500, color: 'lightgray', ...Style.noWrapEllipsis } }, ['Result: View cache diff'])
+    ])
+  }
+  const step2 = () => {
+    const otherCallIndexSelector = (metadata, fqn) => {
+      const shards = _.flow(
+        _.map(c => Number(c.shardIndex)),
+        _.uniq
+      )(metadata.calls[fqn])
+
+      if (shards.length === 1) {
+        if (otherIndex !== shards[0]) { setOtherIndex(shards[0]) }
+        if (shards[0] === -1) {
+          return 'N/A (this call was not scattered)'
+        } else {
+          return `${shards[0]} (exactly one shard in the scatter)`
+        }
+      } else {
+        const shardOptions = _.map(i => { return { value: i, label: i } }, shards)
+        return h(Select, { options: shardOptions, onChange: i => { setOtherIndex(i.value) } })
+      }
+    }
+
+    const otherCallAttemptDecision = (metadata, fqn, index) => {
+      const successfulAttempt = _.flow(
+        _.filter(c => c.shardIndex === index && c.executionStatus === 'Done'),
+        _.map(c => c.attempt),
+        _.first
+      )(metadata.calls[fqn])
```

Comment:
Since we just want one, we could use `_.find` to simplify and save some work.
```suggestion
      const successfulAttempt = _.find({ shardIndex: index, executionStatus: 'Done' }, metadata.calls[fqn])?.attempt
```

Reply:
👍  and in fact I managed to simplify even further. I don't think users will care which attempt succeeded, only whether the shard eventually succeeded or not.